parquet-hadoop/src/test/java/parquet/filter2/statisticslevel/TestStatisticsFilter.java

```diff
@@ -279,10 +279,10 @@ public void testUdp() {
   @Test
   public void testClearExceptionForNots() {
     List<ColumnChunkMetaData> columnMetas = Arrays.asList(
-        getIntColumnMeta(new IntStatistics(), 0L),
-        getDoubleColumnMeta(new DoubleStatistics(), 0L));
+        getDoubleColumnMeta(new DoubleStatistics(), 0L),
```

Comment:
This test case starts failing because earlier, when we did not exit with stats.isEmpty() , filter condition will be evaluated for the IntColumn, and will return true, now since for the 'and' predicate we have : 

@Override
  public Boolean visit(And and) {
    return and.getLeft().accept(this) && and.getRight().accept(this);
  }

It will try to filter for double column and get an exception. 

With the stats.isEmpty() check in place, IntColumn filter will itself return a false (since the test is using an empty stats object), there would be no need to evaluate the doubleColumn filter check, and hence no exception contrary to what we are expecting !